### PR TITLE
Add nested loops example

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #

**Description of your changes:**
This is an example to demonstrate how to use loops in KFP-Tekton and the complexity loops can be. This example also highlights the limitation of the current Tekton API and the importance to have custom task spec. 

**Environment tested:**

* Python Version (use `python --version`): 3.8
* Tekton Version (use `tkn version`): 0.21
* Kubernetes Version (use `kubectl version`): 1.18
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

**Comment by fenglixa** [comment]
Thanks @Tomcli ,  Is there any difference of this PR with the nested loop in test_data?
https://github.com/kubeflow/kfp-tekton/blob/master/sdk/python/tests/compiler/testdata/withitem_nested.py

**Comment by Tomcli** [comment]
> Thanks @Tomcli , Is there any difference of this PR with the nested loop in test_data?
> https://github.com/kubeflow/kfp-tekton/blob/master/sdk/python/tests/compiler/testdata/withitem_nested.py

yes, that's the same one. I try to get more users to test the loop features, so I want to give some clear instructions on how to use it and see what are the gaps we have.

As part of this I figured out some changes in kfp v1.4.0 is breaking the versioning for loop as kfp 1.4.0 no longer provides the uuid for the loop opsgroup
#485

File: samples/README.md [hunk 1]
```diff
@@ -6,13 +6,7 @@ If you are interested more in the larger list of pipelines samples we are testin
...
 
 ## Prerequisites
-- Install [Kubeflow 1.0.2+](https://www.kubeflow.org/docs/started/getting-started/) and connect the cluster to the current shell with `kubectl`
-- Install [Tekton 0.16.3](https://github.com/tektoncd/pipeline/releases/tag/v0.16.3)
-    - For KFP, we shouldn't be modifying the default work directory for any component. Therefore, please run the below command to disable the [home and work directories overwrite](https://github.com/tektoncd/pipeline/blob/master/docs/install.md#customizing-the-pipelines-controller-behavior) from Tekton default.
-        ```shell
-        kubectl patch cm feature-flags -n tekton-pipelines -p '{"data":{"disable-home-env-overwrite":"true","disable-working-directory-overwrite":"true"}}'
-        ```
-- Install [Kubeflow pipeline with Tekton backend](/guides/kfp_tekton_install.md)
+- Install [Kubeflow 1.3+](https://www.kubeflow.org/docs/started/getting-started/) or install [standalone kfp-tekton 0.7.0+](/guides/kfp_tekton_install.md#standalone-kubeflow-pipelines-with-tekton-backend-deployment). Then connect the cluster to the current shell with `kubectl`
```
Comment (ckadner) [inline samples/README.md:9]: The first part of this sentence before the "or" makes it sound like users can choose to only install Kubeflow (without Tekton or the KFP-Tekton backend)?

**Review by ckadner — CHANGES_REQUESTED** [review]

File: samples/README.md (hunk 1 above)
Comment (Tomcli) [reply samples/README.md]: I updated the wording to ask users to deploy kfp with tekton backend.

**Review by ckadner — APPROVED** [review]
thanks Tommy

/lgtm

**Pull request merged**